api/tests/Behat/bootstrap/v2/Common/ExpectedResultsTrait.php

```diff
@@ -0,0 +1,158 @@
+                    $listItems = $tableItem->findAll('xpath', $xpath);
+
+                    if (count($listItems) > 0) {
+                        $this->getTextItems($listItems);
+                    } else {
+                        $xpath = '//dt|//dd';
+                        $descriptionDataItems = $tableItem->findAll('xpath', $xpath);
+                        $this->getTextItems($descriptionDataItems);
+                    }
+                }
+            } elseif ('tbody' == $tableItem->getTagName()) {
+                $xpath = '//tr';
+                $tableRowItems = $tableItem->findAll('xpath', $xpath);
+
+                if (count($tableRowItems) > 0) {
+                    foreach ($tableRowItems as $tableRowItem) {
+                        $xpath = '//td|//th';
+                        $tableDataItems = $tableRowItem->findAll('xpath', $xpath);
+                        $this->getTextItems($tableDataItems);
+                    }
+                } else {
+                    $xpath = '//td|//th';
+                    $tableDataItems = $tableItem->findAll('xpath', $xpath);
+                    $this->getTextItems($tableDataItems);
+                }
+            } else {
+                $this->throwContextualException('Unrecognised option');
+            }
+            $sections[] = $this->comparisonItems;
+        }
+
+//        ** Uncomment below for debugging purposes... useful when creating your tests**
+//        foreach ($sections as $sectionKey=>$section) {
+//            var_dump("Section key: ".strval($sectionKey));
+//            var_dump($section);
+//        }
+//        var_dump($expectedResultArrays);
+
+        foreach ($sections[$sectionNumber] as $foundResultKey => $foundResults) {
+            $this->checkContainsExpectedResults($expectedResultArrays[$foundResultKey], $foundResults, $sectionNumber, $context);
+        }
+    }
+
+    private function getTextItems($items)
+    {
+        $tableValues = [];
+        foreach ($items as $item) {
+            $tableValues[] = trim(strval($item->getText()));
+        }
+        $this->comparisonItems[] = $tableValues;
+    }
+
+    private function checkContainsExpectedResults($expectedItems, $foundItems, $sectionNumber, $context)
+    {
+        $foundInElemPrevious = 0;
+        $raiseException = false;
+        foreach ($expectedItems as $expectedItem) {
+            $found = false;
+            foreach (array_slice($foundItems, $foundInElemPrevious) as $foundItemKey => $foundItem) {
+                if (str_contains(strval(trim(strtolower($foundItem))), strval(trim(strtolower($expectedItem))))) {
+                    $found = true;
+                    $foundInElem = $foundItemKey + $foundInElemPrevious;
+                    break;
+                }
+            }
+            if ($found and $foundInElem >= $foundInElemPrevious) {
+                $foundInElemPrevious = $foundInElem;
+            } else {
+                $raiseException = true;
+            }
```

Comment:
Finding this quite hard to follow - could you take me through it?

Reply:
To be honest it was hard to implement lol. Broke my brain a bit... I'll talk it through with you